tests/tests/util/web3Requests.ts

```diff
@@ -0,0 +1,125 @@
+import Web3 from "web3";
+import { JsonRpcResponse } from "web3-core-helpers";
+import { TransactionReceipt } from "web3-core";
+import { AbiItem } from "web3-utils";
+import { Contract } from "web3-eth-contract";
+import { GENESIS_ACCOUNT, GENESIS_ACCOUNT_PRIVATE_KEY } from "../constants";
+
+// make a web3 request, adapted to manual seal testing
+export async function customRequest(web3: Web3, method: string, params: any[]) {
+  return new Promise<JsonRpcResponse>((resolve, reject) => {
+    (web3.currentProvider as any).send(
+      {
+        jsonrpc: "2.0",
+        id: 1,
+        method,
+        params,
+      },
+      (error: Error | null, result?: JsonRpcResponse) => {
+        if (error) {
+          reject(
+            `Failed to send custom request (${method} (${params.join(",")})): ${
+              error.message || error.toString()
+            }`
+          );
+        }
+        resolve(result);
+      }
+    );
+  });
+}
+
+// wrap the above function to catch errors and return them into a JsonRpc format
+export async function wrappedCustomRequest(
+  web3: Web3,
+  method: string,
+  params: any[]
+): Promise<JsonRpcResponse> {
+  try {
+    let resp = await customRequest(web3, method, params);
+    return resp;
+  } catch (e) {
+    return {
+      jsonrpc: "req error",
+      id: 0,
+      error: typeof e === "string" ? e.toString() : JSON.stringify(e),
+    };
+  }
+}
+
+// Create a block and finalize it.
+// It will include all previously executed transactions since the last finalized block.
+export async function createAndFinalizeBlock(web3: Web3): Promise<number> {
```

Comment:
This should be moved to the PolkadotJs library, as it is a substrate rpc call, not eth

Reply:
>This should be moved to the PolkadotJs library

do you mean create a separate file for non-web3 related requests?

Reply:
Since we now have the polkadotjs library included in the test, I think it should be moved to the file taking care of it